docs/src/props-table.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import {Box, Link, Label} from '@primer/components'
+import Table from '@primer/gatsby-theme-doctocat/src/components/table'
+
+function PropsTable({children}) {
+  return (
+    <Table>
+      <thead>
+        <tr>
+          <Box as="th" textAlign="left">
+            Name
+          </Box>
+          <Box as="th" textAlign="left">
+            Type
+          </Box>
+          <Box as="th" textAlign="left">
+            Default
+          </Box>
+          <Box as="th" textAlign="left">
+            Description
+          </Box>
+        </tr>
+      </thead>
+      <tbody>{children}</tbody>
+    </Table>
+  )
+}
+
+function Row({name, type, defaultValue, description, required}) {
+  return (
+    <tr>
+      <Box as="td" fontFamily="mono" fontSize={1} fontWeight="bold" sx={{whiteSpace: 'nowrap'}}>
+        {typeof name === 'function' ? name() : name}
+        {required ? (
+          <>
+            {` `}
+            <Label style={{verticalAlign: 'text-top'}}>Required</Label>
+          </>
+        ) : null}
+      </Box>
+      <Box as="td" fontFamily="mono" fontSize={1}>
+        {typeof type === 'function' ? type() : type}
+      </Box>
+      <Box as="td" fontFamily="mono" fontSize={1}>
+        {typeof defaultValue === 'function' ? defaultValue() : defaultValue}
+      </Box>
+      <td>{typeof description === 'function' ? description() : description}</td>
+    </tr>
+  )
+}
+
+function AsRow({defaultValue}) {
+  return (
+    <Row
+      name="as"
+      defaultValue={defaultValue}
+      type="ReactElement"
```

Comment:
Is this type correct?

I think the type definition for `as` was something like `keyof JSX.IntrinsicElements | React.ComponentType`

Also, I'm wondering if we should allow people to override the type so they restrict the values that can be passed to `as`

Reply:
Ahh I think you're right, and there's `React.ElementType`: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L73 as shorthand (with some other special stuff we don't need?)